src/ts/ConvivaAnalyticsTracker.ts

```diff
@@ -0,0 +1,939 @@
+
+    this.logger = new Html5Logging();
+    this.sessionKey = Conviva.Constants.NO_SESSION_KEY;
+    this._isAdBreakActive = false;
+
+    const deviceMetadataFromConfig = this.config.deviceMetadata || {};
+    const deviceMetadata: Conviva.ConvivaDeviceMetadata = {
+      [Conviva.Constants.DeviceMetadata.CATEGORY]:
+        deviceMetadataFromConfig.category || this.config.deviceCategory || Conviva.Constants.DeviceCategory.WEB,
+      [Conviva.Constants.DeviceMetadata.BRAND]: deviceMetadataFromConfig.brand,
+      [Conviva.Constants.DeviceMetadata.MANUFACTURER]: deviceMetadataFromConfig.manufacturer,
+      [Conviva.Constants.DeviceMetadata.MODEL]: deviceMetadataFromConfig.model,
+      [Conviva.Constants.DeviceMetadata.TYPE]: deviceMetadataFromConfig.type,
+      [Conviva.Constants.DeviceMetadata.VERSION]: deviceMetadataFromConfig.version,
+      [Conviva.Constants.DeviceMetadata.OS_NAME]: deviceMetadataFromConfig.osName,
+      [Conviva.Constants.DeviceMetadata.OS_VERSION]: deviceMetadataFromConfig.osVersion,
+    };
+    Conviva.Analytics.setDeviceMetadata(deviceMetadata);
+
+    let callbackFunctions: Record<string, Function> = {};
+    callbackFunctions[Conviva.Constants.CallbackFunctions.CONSOLE_LOG] = this.logger.consoleLog;
+    callbackFunctions[Conviva.Constants.CallbackFunctions.MAKE_REQUEST] = new Html5Http().makeRequest;
+    const html5Storage = new Html5Storage();
+    callbackFunctions[Conviva.Constants.CallbackFunctions.SAVE_DATA] = html5Storage.saveData;
+    callbackFunctions[Conviva.Constants.CallbackFunctions.LOAD_DATA] = html5Storage.loadData;
+    callbackFunctions[Conviva.Constants.CallbackFunctions.CREATE_TIMER] = new Html5Timer().createTimer;
+    callbackFunctions[Conviva.Constants.CallbackFunctions.GET_EPOCH_TIME_IN_MS] = new Html5Time().getEpochTimeMs;
+
+    const settings: Record<string, string | number> = {};
+    settings[Conviva.Constants.GATEWAY_URL] = config.gatewayUrl;
+    settings[Conviva.Constants.LOG_LEVEL] = this.config.debugLoggingEnabled
+      ? Conviva.Constants.LogLevel.DEBUG
+      : Conviva.Constants.LogLevel.NONE;
+
+    Conviva.Analytics.init(customerKey, callbackFunctions, settings);
+
+    this.contentMetadataBuilder = new ContentMetadataBuilder(this.logger);
+
+    this.registerPlayerEvents();
+  }
+
+  /**
+   * Initializes a new conviva tracking session.
+   *
+   * Warning: The integration can only be validated without external session managing. So when using this method we can
+   * no longer ensure that the session is managed at the correct time. Additional: Since some metadata attributes
+   * relies on the players source we can't ensure that all metadata attributes are present at session creation.
+   * Therefore it could be that there will be a 'ContentMetadata created late' issue after conviva validation.
+   *
+   * If no source was loaded and no assetName was set via updateContentMetadata this method will throw an error.
+   */
```

Comment:
I think all these documentation blocks for methods actually belong to `ConvivaAnalytics.ts` as this is for the user of the integration.

Reply:
Moved in https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/112/commits/5803fb7253bcc383ceaec4f74a6d89255e1f09e1.